docs/src/verification.md

```diff
@@ -0,0 +1,166 @@
+# Verification in tests
+
+## General Overview
+
+When comparing our `compiled_model` with the `framework_model` (e.g., `PyTorch`), we aim to verify whether the output from the `compiled_model` is sufficiently similar to the output from the `framework_model` (where required degree of similarity is configurable).
+
+So generally we want to perform the following steps:
+1. Create a framework model
+2. Run a forward pass through the framework model
+3. Compile the framework model using `Forge`
+4. Run a forward pass through the compiled model
+5. Compare the outputs.
+
+Most of the above steps `verify()` function does for us:
+- Handles forward passes for both framework and compiled models
+- Compares results using a combination of comparison methods
+- Supports customization through the `VerifyConfig` class.
+
+**Example of usage**
+```python
+def test_add():
+
+    class Add(nn.Module):
+        def __init__(self):
+            super().__init__()
+
+        def forward(self, a, b):
+            return a + b
+
+
+    inputs = [torch.rand(2, 32, 32), torch.rand(2, 32, 32)]
+
+    framework_model = Add()
+    compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+
+    verify(inputs, framework_model, compiled_model)
+```
+
+**Notes:**
+- If you only want to compile model and perform forward pass without comparing outputs you can just:
+```python
+def test_add():
+
+    class Add(nn.Module):
+        def __init__(self):
+            super().__init__()
+
+        def forward(self, a, b):
+            return a + b
+
+
+    inputs = [torch.rand(2, 32, 32), torch.rand(2, 32, 32)]
+
+    framework_model = Add()
+    compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+
+    fw_out = framework_model(*inputs)
+    co_out = compiled_model(*inputs)
+```
+
+## Verify Config Overview
+
+If `VerifyConfig` isn't passed as a param, default one will be used. Currently through `VerifyConfig` you can disable/enable:
+
+| Feature                           | Name                | Enabled (default) |
+|-----------------------------------|---------------------|:-----------------:|
+| Verification as a method          | `enabled`           | `True`            |
+| Number of output tensors check    | `verify_size`       | `True`            |
+| Output type check                 | `verify_dtype`      | `True`            |
+| Output shape check                | `verify_shape`      | `True`            |
+
+
+For more information about `VerifyConfig` you can check `forge/forge/verify/config.py`.
+
+**Example of usage**
+```python
```

Comment:
Maybe we can keep following examples a bit shorter. E.g. now that we have full example above, we can do something like this:
```py
    ...

    fw_out = framework_model(*inputs)
    co_out = compiled_model(*inputs)

    verify(inputs, framework_model, compiled_model, VerifyConfig(verify_dtype=False))
```